### PR TITLE
Review and update Upgrade doc

### PR DESCRIPTION
**Pull request by lread**
Done:
1. Don't immediately presume macOS.
2. Add a little generic text, from which I linked to other useful docs.
3. Match same install variations from Install doc
4. Add toc to match Install doc
5. Separate Linux and Windows (the grouping was convenient for author but not the reader).

Not done:
1. The is no mention of SNAPSHOT builds. A reader might wonder how to upgrade (not sure if this is even do-able for brew and nix):
   1. from a final release to a SNAPSHOT release
   2. from a SNAPSHOT release to a latest SNAPSHOT release under same version

### CODE REVIEW

File: doc/upgrade.adoc [hunk 1]
```diff
@@ -1,18 +1,59 @@
...
+
+Followed by:
+[source,shell]
+----
+nix-env -u polylith
+----
+
+== As Dependency
+
+Update your `polylith/clj-poly` dependency in your `deps.edn` to the current version.
```
Comment (tengstrand) [inline doc/upgrade.adoc]: I have a habit of writing `./deps.edn` if we refer to the deps.edn file at the workspace root (bricks and projects have their own).

**Review by tengstrand — CHANGES_REQUESTED** [review]
Made one comment, but except for that, it looks great with many improvements!

**Comment by tengstrand** [comment]
I think it's fine that this page only explains how to upgrade the stand-alone poly tool. I don't know yet how to upgrade a stand-alone pre-release of the poly tool, but perhaps we can revisit this page when we have made our first SNAPSHOT release.

File: doc/upgrade.adoc (hunk 1 above)
Comment (lread) [reply doc/upgrade.adoc]: Ya, that's a good convention, will do!

**Comment by lread** [comment]
> I think it's fine that this page only explains how to upgrade the stand-alone poly tool.

To make sure I understand, are you suggesting I remove sections?:
- As Dependency
- Clojure CLI Tool

> I don't know yet how to upgrade a stand-alone pre-release of the poly tool, but perhaps we can revisit this page when we have made our first SNAPSHOT release.

Sounds good.

**Comment by tengstrand** [comment]
> To make sure I understand, are you suggesting I remove sections?:
No, I was unclear. It's fine as it is!

**Comment by lread** [comment]
@tengstrand, good to merge, I think.

**Review by tengstrand — APPROVED** [review]

**Pull request merged**